### PR TITLE
Update allsorts and add validate command

### PR DESCRIPTION
**Pull request by wezm**
This PR:

* Updates to allsorts to 0.2
* Adds ~~`parse`~~ `validate` command
* Includes a glyph count in the `dump` command output for TTF fonts

The `validate` command attempts to parse all the glyphs (or various DICTs in the case of CFF) in the supplied font. It reports any errors encountered but is otherwise silent. This command was useful for bulk testing `allsorts` against a repertoire of real world fonts.

### CODE REVIEW

File: src/cli.rs [hunk 1]
```diff
@@ -14,6 +14,9 @@ pub enum Command {
     #[options(help = "dump font information")]
     Dump(DumpOpts),
 
+    #[options(help = "parse the supplied font, reporting any failures")]
+    Parse(ParseOpts),
```
Comment (brendanzab) [inline src/cli.rs]: Would `validate` be a better name? `parse` seems to imply that it would produce some output. 🤔

File: src/parse.rs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+    let cff = scope.read::<CFF>()?;
+    if cff.name_index.count != 1 {
+        return Err(ParseError::BadIndex);
+    }
+    let font = cff.fonts.get(0).ok_or(ParseError::MissingValue)?;
+    let char_strings_offset = font
+        .top_dict
+        .get_i32(Operator::CharStrings)
+        .ok_or(ParseError::MissingValue)??;
```
Comment (brendanzab) [inline src/parse.rs]: lol `??`? 🤔

Comment (wezm) [reply src/parse.rs]: `get_i32` returns `Option<Result>`, `ok_or` makes it `Result<Result>`, `??` unwraps :upside_down_face:

File: src/cli.rs (hunk 1 above)
Comment (wezm) [reply src/cli.rs]: I like it! That does seem like a better name.

Comment (wezm) [reply src/cli.rs]: Rename done.

**Review by brendanzab — APPROVED** [review]

**Comment by brendanzab** [comment]
Looks good @wezm!

**Pull request merged**